adapters/src/main/kotlin/com/krzykrucz/elesson/currentlesson/lessonprogress/adapters/rest/LessonProgressRequestHandler.kt

```diff
@@ -0,0 +1,60 @@
+package com.krzykrucz.elesson.currentlesson.lessonprogress.adapters.rest
+
+import arrow.core.Option
+import arrow.core.extensions.fx
+import arrow.core.getOrElse
+import arrow.core.none
+import arrow.core.some
+import com.krzykrucz.elesson.currentlesson.infrastructure.run
+import com.krzykrucz.elesson.currentlesson.lessonprogress.adapters.persistence.createLessonProgressView
+import com.krzykrucz.elesson.currentlesson.shared.ClassName
+import com.krzykrucz.elesson.currentlesson.shared.LessonHourNumber
+import com.krzykrucz.elesson.currentlesson.shared.LessonIdentifier
+import com.krzykrucz.elesson.currentlesson.shared.NonEmptyText
+import org.springframework.web.reactive.function.BodyInserters
+import org.springframework.web.reactive.function.server.ServerRequest
+import org.springframework.web.reactive.function.server.ServerResponse
+import reactor.core.publisher.Mono
+import java.time.LocalDate
+import java.util.*
+
+fun handleLessonProgressViewRequest(serverRequest: ServerRequest): Mono<ServerResponse> {
+    val readDateFromParams = serverRequest.queryParam("date").toOption()
+        .map { LocalDate.parse(it) }
+
+    val readLessonHourNumberFromParams = serverRequest.queryParam("lessonHourNumber").toOption()
+        .flatMap { LessonHourNumber.of(it.toInt()) }
+
+    val readClassNameFromParams = serverRequest.queryParam("className").toOption()
+        .map { ClassName(NonEmptyText.of(it)!!) }
```

Comment:
is it safe to invoke `!!` here?

Reply:
Changed it to NonEmptyText constructor. Nevertheless, we had Option<String> there to map over so `it` was a non-null String.